var/spack/repos/builtin/packages/magma/package.py

```diff
@@ -150,6 +151,7 @@ def cmake_args(self):
             # See https://github.com/ROCm/rocFFT/issues/322
             if spec.satisfies("^cmake@3.21.0:3.21.2"):
                 options.append(define("__skip_rocmclang", True))
+            options.append(define("ROCM_CORE", spec["rocm-core"].prefix))
```

Comment:
This is causing CI to fail.  This needs to reference the path to the correct dependency.

Reply:
Which CI?  I don't see a failure.  Does rocm-core have to be a direct dependency of the package in order to be used?  It is a dependency of the hip package.

Reply:
OK, so I think it would fail if the hip package is an external.  Correct? I would then just need to make rocm-core a direct dep of magma.

Reply:
It's unfortunate we don't make this clearer in more places in the documentation but you are correct about using `self.spec` object to query direct dependencies.  We touch on it in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#accessing-dependencies.